fission-cli/library/Fission/CLI/Parser/Command/App/Init/Types.hs

```diff
@@ -2,18 +2,18 @@ module Fission.CLI.Parser.Command.App.Init.Types (Options (..)) where
 
 import           Fission.Prelude
 
-import qualified Fission.URL                          as URL
+import qualified Fission.App.Name                     as App
 
 import qualified Fission.CLI.Parser.Config.IPFS.Types as IPFS
 import           Fission.CLI.Parser.Verbose.Types
 
 -- | Arguments, flags & switches for the `app init` command
 data Options = Options
-  { appDir       :: FilePath
-  , buildDir     :: Maybe FilePath
-  , maySubdomain :: Maybe URL.Subdomain
-  , ipfsCfg      :: IPFS.Config
-  , verboseFlag  :: VerboseFlag -- ^ Verbose flag
+  { appDir      :: FilePath
+  , buildDir    :: Maybe FilePath
+  , mayAppName  :: Maybe App.Name
+  , ipfsCfg     :: IPFS.Config
+  , verboseFlag :: VerboseFlag -- ^ Verbose flag
   } deriving (Show, Eq)
```

Comment:
Since we change the API to enforce the rules, we also validate this on the CLI to make sure we're passing valid data. This isn't only a good practice so we can validate CLI-side, but the `Web.API` types also enforce it so this change is nonoptional